### DISCUSSION TITLE
[Help] Where can I get Cairo?

### DISCUSSION THREAD

**Discussion by ZiguoAtGitHub** [cduck/drawSvg#86] [Q&A] [answered]
Hi there,
My env is win10(x64)+python3.11.1(x64). I just install drawSvg through pip. 
But I can not run the example script because of no Cairo. Where can I get the Cairo to install and how can I run the examples correctly?
Thank you all!

Here are the errors.
```
Original OSError: no library called "cairo-2" was found
no library called "cairo" was found
no library called "libcairo-2" was found
cannot load library 'libcairo.so.2': error 0x7e
cannot load library 'libcairo.2.dylib': error 0x7e
cannot load library 'libcairo-2.dll': error 0x7e
  warnings.warn(msg, RuntimeWarning)
```

**Comment by cduck** [comment, marked as answer]
Note: Cairo is not required to use most features of drawsvg.  Remove lines `d.rasterize()` and `d.save_png()` from the example code.  Use `d.save_svg()` instead.

I'm not sure the best way to install Cairo on Windows, that's why only the Linux and Mac installation instructions are listed on the README.  You can try the instructions here: https://www.cairographics.org/download/.  If you succeed, please let me know what worked and I'll add the instructions to the README.

Edit:
Updated for drawsvg 2.0.  For anyone else having trouble installing Cairo, [try the conda package manager](https://github.com/cduck/drawSvg#full-feature-install).

**Reply by ZiguoAtGitHub** [reply]
Hi Casey
Thanks for your fast reply.

I did try to install all the files which can be downloaded from https://gladewin32.sourceforge.net/ linked from https://www.cairographics.org/download/ and ther is no surprise.
I think maybe the version of Glade and Gtk are too old.

I run the example code named "Animation with the SVG Animate Tag" and there is no the two lines you noted. But the problem is still.
```
Original OSError: no library called "cairo-2" was found
no library called "cairo" was found
no library called "libcairo-2" was found
cannot load library 'libcairo.so.2': error 0x7e
cannot load library 'libcairo.2.dylib': error 0x7e
cannot load library 'libcairo-2.dll': error 0x7e
  warnings.warn(msg, RuntimeWarning)

```
As you noted, Could drawSvg get rid off the prerequisite of Cairo totally?
:)

**Reply by cduck** [reply]
It's unfortunate the Windows installation out of date.  Luckily, Cairo will soon be an optional dependency once drawsvg version 2 is out (#89).

To be clear.  Is this error message causing the code to stop or is it just a warning message?  You should be able to ignore the warning and continue to use the SVG functionality.

**Reply by ZiguoAtGitHub** [reply]
> It's unfortunate the Windows installation out of date. Luckily, Cairo will soon be an optional dependency once drawsvg version 2 is out (#89).
> 
> To be clear. Is this error message causing the code to stop or is it just a warning message? You should be able to ignore the warning and continue to use the SVG functionality.

Unfortunately, the code stops due to the errors.

**Reply by cduck** [reply]
That's not good.  It's supposed to be a warning.  Please try out the new pre-release of version 2 by following the install instructions in #89 and let me know if it works for you.  You'll have to get new example code from the updated readme page.

**Reply by ZiguoAtGitHub** [reply]
Hi
I tried twice as you mentioned but failed in installing the pre-release of drawSvg 2.0.

The first try is just copy and paste the command "python3 -m pip install -e "git+https://github.com/cduck/drawsvg.git@v2#egg=drawsvg[all]" but there is no response 
The second is changing the python3 to python because of no python3.exe in the python3.11 directory.

My knowledge about python is not enough to solve the problem. Would you give me some hints? 

Here is the information :

> C:\Users\xuzig\AppData\Local\Programs\Python\Python311>python3 -m pip install -e "git+https://github.com/cduck/drawsvg.git@v2#egg=drawsvg[all]"
> 
> C:\Users\xuzig\AppData\Local\Programs\Python\Python311>python -m pip install -e "git+https://github.com/cduck/drawsvg.git@v2#egg=drawsvg[all]"
> Obtaining drawsvg[all] from git+https://github.com/cduck/drawsvg.git@v2#egg=drawsvg[all]
>   Cloning https://github.com/cduck/drawsvg.git (to revision v2) to c:\users\xuzig\appdata\local\programs\python\python311\src\drawsvg
>   ERROR: Error [WinError 2] 系统找不到指定的文件。 while executing command git version
> ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?
> 
> [notice] A new release of pip available: 22.3.1 -> 23.0
> [notice] To update, run: python.exe -m pip install --upgrade pip
>

**Reply by cduck** [reply]
That installation command uses git, a common source code management tool.  Use this other command that doesn't need git:
`python.exe -m pip install https://github.com/cduck/drawsvg/archive/v2.zip`

(Alternative: install git from https://git-scm.com/download/win)

**Reply by ZiguoAtGitHub** [reply]
I installed the new version of drawsvg successfully by following your instruction and got the svg file created by example code "Animation with the SVG Animate Tag". Thanks!
There is no any other warnings or errors except the some name-changes in new verison drawsvg, e.g. the class name (drawSvg change to drawsvg) and some functions' name (e.g. appendAnim change to append_anim).
Good job!👍🙂

**Reply by cduck** [reply]
Great!  Thanks for testing it out.
